dateutil/test/test_tz.py

```diff
@@ -1079,6 +1082,28 @@ def testGettzCacheTzLocal(self):
         assert local1 is not local2
 
 
+@pytest.mark.gettz
+def test_gettz_same_result_for_none_and_empty_string():
+    local_from_none = tz.gettz()
+    local_from_empty_string = tz.gettz("")
+    assert local_from_none is not None
+    assert local_from_empty_string is not None
+    assert local_from_none == local_from_empty_string
+
+
+@pytest.mark.gettz
+@pytest.mark.parametrize("gettz_arg", [None, ""]) # could add colon here
+@given(dt=st.datetimes(max_value=datetime.now()))
```

Comment:
Without setting a `max_value` the tests fail for me because they return standard and not summer time. Could this maybe be expected for dates further into the future (e.g. `datetime(2038, 4, 1, 0, 0)`) due to different data sources of `tzlocal` and `gettz`? I'm not sure about the whole timezone stuff yet.

Reply:
Since `datetime` bounds have to be naive I instead initialized the constant from a timestamp.

The test now also has a minimum bound, since I ran into an issue when `dt.astimezone()` started returning LMT (solar time?) in the 19th century.

This looks like a similar issue to the one with the upper bound, although I assume the problem in the case is probably that my local zone file starts at 1901 (`zdump -v /etc/localtime | head -n 1`), after the transition from LMT was already over, while `astimezone()` gets more historical information?

If that is the case I am not sure which date would be an appropriate lower bound, other than 1970.

Reply:
> The test now also has a minimum bound, since I ran into an issue when `dt.astimezone()` started returning LMT (solar time?) in the 19th century.

This is what it's supposed to do. LMT = Local Mean Time. I think you may be right that this is again a Version 1 vs. Version 2 issue. I believe the appropriate lower bound is something close to the negative equivalent of the epochalypse: `1901-12-13T20:45:52`.

If it works within that range (or even within a range starting ~24 hours after that), that is probably best.

1970-01-01 is also an *OK* lower bound in the sense that the TZ database doesn't make any guarantees about data quality before that, but I think that if we see any deviations in timestamps within the range `[-2**31, 2**31]` between `.astimezone()` and `tz.gettz()`, it may be because the two are using different data sources and not just different implementations reading the same data source; in that case, we may need to scrap the whole idea.

Reply:
I've just checked and you are correct, the first date in my zone file is in fact the negative epochalypse. I guess that's another date to keep in mind for when I run into time zone issues.

I had to initialize it with `timedelta` since I got a `ValueError` on Windows when initializing with `fromtimestamp()`.
```
ValueError: timestamp out of range for platform localtime()/gmtime() function
```